Codecademy/client-modules
# feat: Minimally added focus-visible with postcss-focus-visible

## Overview
Per the [:focus-visible Polyfill RFC](https://www.notion.so/codecademy/focus-visible-Polyfill-af0e840981c74f869ee2dbf63d5b483f), brings in [`postcss-focus-visible`](https://www.npmjs.com/package/postcss-focus-visible) to allow us to write `:focus-visible` in code and have it be polyfilled.

Does not bring in the [`focus-visible`](https://www.npmjs.com/package/focus-visible) polyfill because that [crashes in SSR](https://codecademy.slack.com/archives/C047ESP1T/p1604700402178800?thread_ts=1604699939.178500&cid=C047ESP1T). We'll need to move that into downstream repos.

### PR Checklist

- [x] Related to JIRA ticket: WEB-1146

### Description

Sets it up in our postcss styles processing our webpack config to only rely on the `[data-focus-visible-added]` selector (instead of duplicating styles).

### Testing Instructions

In each of a browser that supports `:focus-visible` (e.g. Chrome) and one that doesn't (e.g. Safari):

1. Visit the (deprecated) Button story in the styleguide
2. Click on a button with your mouse -> you should not see the focus outline
3. <kbd>Tab</kbd> to the button with your keyboard -> you should see the focus outline

[Enabling keyboard navigation in Mac OS X Web browsers](https://www.456bereastreet.com/archive/200906/enabling_keyboard_navigation_in_mac_os_x_web_browsers/#:~:text=Choose%20Safari%20%3E%20Preferences.,each%20item%20on%20a%20webpage.)
